### PR TITLE
refactor: Expose bindplane server & Rest funcs

### PR DESCRIPTION
**Pull request by Mrod1598**
### Proposed Change
- Move `Bindplane` interface into exported server package
- Create interface for `Relayers` for different implementation in ee
- Expose `rest` api functions for use downstream

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: cli/commands/serve/server.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ import (
 	"github.com/observiq/bindplane-op/otlp"
 	"github.com/observiq/bindplane-op/resources"
 	"github.com/observiq/bindplane-op/rest"
+	exposedServer "github.com/observiq/bindplane-op/server"
```
Comment (cpheps) [inline cli/commands/serve/server.go:43]: Well explicitly called out for package aliases [Effective Go](https://go.dev/doc/effective_go#package-names) does call out package names should be lower cased. Can we lower case this to just `exposedserver`?

File: internal/graphql/api.go [hunk 2]
```diff
@@ -27,11 +27,11 @@ import (
 	"github.com/gorilla/websocket"
 
 	"github.com/observiq/bindplane-op/internal/graphql/generated"
-	"github.com/observiq/bindplane-op/internal/server"
+	exposedServer "github.com/observiq/bindplane-op/server"
```
Comment (cpheps) [inline internal/graphql/api.go:30]: Same here

File: internal/graphql/overview.go [hunk 3]
```diff
@@ -21,13 +21,13 @@ import (
 	"strings"
 
 	model1 "github.com/observiq/bindplane-op/internal/graphql/model"
-	"github.com/observiq/bindplane-op/internal/server"
 	"github.com/observiq/bindplane-op/model"
 	"github.com/observiq/bindplane-op/model/graph"
 	"github.com/observiq/bindplane-op/model/otel"
+	exposedServer "github.com/observiq/bindplane-op/server"
```
Comment (cpheps) [inline internal/graphql/overview.go:27]: And here

**Review by cpheps — COMMENTED** [review]
Looks good will approve after fixup of package alias.

**Review by cpheps — APPROVED** [review]

File: rest/error.go [hunk 4]
```diff
@@ -28,7 +28,8 @@ func NewErrorResponse(err error) ErrorResponse {
 	return ErrorResponse{Errors: []string{err.Error()}}
 }
 
-func handleErrorResponse(c *gin.Context, statusCode int, err error) {
+// HandleErrorResponse handles an error response
```
Comment (antonblock) [inline rest/error.go:31]:
```suggestion
// HandleErrorResponse writes an error response and status code to the gin context
```

**Pull request merged**